from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_DetectorWindow(object):
    def setupUi(self, DetectorWindow):
        DetectorWindow.setObjectName("DetectorWindow")
        DetectorWindow.resize(600, 300)
        DetectorWindow.setMinimumSize(QtCore.QSize(600, 300))
        DetectorWindow.setMaximumSize(QtCore.QSize(800, 450))
        font = QtGui.QFont()
        font.setFamily("Bahnschrift SemiBold")
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        DetectorWindow.setFont(font)
        self.horizontalLayout = QtWidgets.QHBoxLayout(DetectorWindow)
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.coordiantes_layout = QtWidgets.QFrame(DetectorWindow)
        self.coordiantes_layout.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.coordiantes_layout.setFrameShadow(QtWidgets.QFrame.Raised)
        self.coordiantes_layout.setObjectName("coordiantes_layout")
        self.verticalLayout = QtWidgets.QVBoxLayout(self.coordiantes_layout)
        self.verticalLayout.setObjectName("verticalLayout")
        self.xcoord_layout = QtWidgets.QHBoxLayout()
        self.xcoord_layout.setObjectName("xcoord_layout")
        self.xcoord_label = QtWidgets.QLabel(self.coordiantes_layout)
        self.xcoord_label.setMaximumSize(QtCore.QSize(16777215, 40))
        self.xcoord_label.setAlignment(QtCore.Qt.AlignCenter)
        self.xcoord_label.setObjectName("xcoord_label")
        self.xcoord_layout.addWidget(self.xcoord_label)
        self.xcoord_box = QtWidgets.QDoubleSpinBox(self.coordiantes_layout)
        self.xcoord_box.setMaximum(999999999.0)
        self.xcoord_box.setSingleStep(0.1)
        self.xcoord_box.setObjectName("xcoord_box")
        self.xcoord_layout.addWidget(self.xcoord_box)
        self.verticalLayout.addLayout(self.xcoord_layout)
        spacerItem = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        self.verticalLayout.addItem(spacerItem)
        self.ycoord_layout = QtWidgets.QHBoxLayout()
        self.ycoord_layout.setObjectName("ycoord_layout")
        self.ycoord_label = QtWidgets.QLabel(self.coordiantes_layout)
        self.ycoord_label.setMaximumSize(QtCore.QSize(16777215, 40))
        self.ycoord_label.setAlignment(QtCore.Qt.AlignCenter)
        self.ycoord_label.setObjectName("ycoord_label")
        self.ycoord_layout.addWidget(self.ycoord_label)
        self.ycoord_box = QtWidgets.QDoubleSpinBox(self.coordiantes_layout)
        self.ycoord_box.setMaximum(999999999.0)
        self.ycoord_box.setSingleStep(0.1)
        self.ycoord_box.setObjectName("ycoord_box")
        self.ycoord_layout.addWidget(self.ycoord_box)
        self.verticalLayout.addLayout(self.ycoord_layout)
        spacerItem1 = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        self.verticalLayout.addItem(spacerItem1)
        self.zcoord_layout = QtWidgets.QHBoxLayout()
        self.zcoord_layout.setObjectName("zcoord_layout")
        self.zcoord_label = QtWidgets.QLabel(self.coordiantes_layout)
        self.zcoord_label.setMaximumSize(QtCore.QSize(16777215, 40))
        self.zcoord_label.setAlignment(QtCore.Qt.AlignCenter)
        self.zcoord_label.setObjectName("zcoord_label")
        self.zcoord_layout.addWidget(self.zcoord_label)
        self.zcoord_box = QtWidgets.QDoubleSpinBox(self.coordiantes_layout)
        self.zcoord_box.setMaximum(999999999.0)
        self.zcoord_box.setSingleStep(0.1)
        self.zcoord_box.setObjectName("zcoord_box")
        self.zcoord_layout.addWidget(self.zcoord_box)
        self.verticalLayout.addLayout(self.zcoord_layout)
        self.horizontalLayout.addWidget(self.coordiantes_layout)
        self.info_layout = QtWidgets.QFrame(DetectorWindow)
        self.info_layout.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.info_layout.setFrameShadow(QtWidgets.QFrame.Raised)
        self.info_layout.setObjectName("info_layout")
        self.verticalLayout_2 = QtWidgets.QVBoxLayout(self.info_layout)
        self.verticalLayout_2.setObjectName("verticalLayout_2")
        self.target_label = QtWidgets.QLabel(self.info_layout)
        self.target_label.setMaximumSize(QtCore.QSize(16777215, 35))
        font = QtGui.QFont()
        font.setPointSize(20)
        self.target_label.setFont(font)
        self.target_label.setAlignment(QtCore.Qt.AlignCenter)
        self.target_label.setObjectName("target_label")
        self.verticalLayout_2.addWidget(self.target_label)
        self.properties_label = QtWidgets.QLabel(self.info_layout)
        self.properties_label.setAlignment(QtCore.Qt.AlignCenter)
        self.properties_label.setObjectName("properties_label")
        self.verticalLayout_2.addWidget(self.properties_label)
        self.properties_layout = QtWidgets.QHBoxLayout()
        self.properties_layout.setObjectName("properties_layout")
        self.width_layout = QtWidgets.QHBoxLayout()
        self.width_layout.setObjectName("width_layout")
        self.width_label = QtWidgets.QLabel(self.info_layout)
        self.width_label.setMaximumSize(QtCore.QSize(16777215, 40))
        self.width_label.setAlignment(QtCore.Qt.AlignCenter)
        self.width_label.setObjectName("width_label")
        self.width_layout.addWidget(self.width_label)
        self.width_box = QtWidgets.QDoubleSpinBox(self.info_layout)
        self.width_box.setMaximum(999999999.0)
        self.width_box.setSingleStep(0.1)
        self.width_box.setProperty("value", 10.0)
        self.width_box.setObjectName("width_box")
        self.width_layout.addWidget(self.width_box)
        self.properties_layout.addLayout(self.width_layout)
        self.height_layout = QtWidgets.QHBoxLayout()
        self.height_layout.setObjectName("height_layout")
        self.height_label = QtWidgets.QLabel(self.info_layout)
        self.height_label.setMaximumSize(QtCore.QSize(16777215, 40))
        self.height_label.setAlignment(QtCore.Qt.AlignCenter)
        self.height_label.setObjectName("height_label")
        self.height_layout.addWidget(self.height_label)
        self.height_box = QtWidgets.QDoubleSpinBox(self.info_layout)
        self.height_box.setMaximum(999999999.0)
        self.height_box.setSingleStep(0.1)
        self.height_box.setProperty("value", 10.0)
        self.height_box.setObjectName("height_box")
        self.height_layout.addWidget(self.height_box)
        self.properties_layout.addLayout(self.height_layout)
        self.verticalLayout_2.addLayout(self.properties_layout)
        spacerItem2 = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
        self.verticalLayout_2.addItem(spacerItem2)
        self.okbutton = QtWidgets.QDialogButtonBox(self.info_layout)
        self.okbutton.setMaximumSize(QtCore.QSize(16777215, 50))
        self.okbutton.setStandardButtons(QtWidgets.QDialogButtonBox.Cancel|QtWidgets.QDialogButtonBox.Ok)
        self.okbutton.setObjectName("okbutton")
        self.verticalLayout_2.addWidget(self.okbutton)
        self.horizontalLayout.addWidget(self.info_layout)

        self.retranslateUi(DetectorWindow)
        QtCore.QMetaObject.connectSlotsByName(DetectorWindow)

    def retranslateUi(self, DetectorWindow):
        _translate = QtCore.QCoreApplication.translate
        DetectorWindow.setWindowTitle(_translate("DetectorWindow", "Detector Settings"))
        self.xcoord_label.setText(_translate("DetectorWindow", "X:"))
        self.xcoord_box.setSuffix(_translate("DetectorWindow", " mm"))
        self.ycoord_label.setText(_translate("DetectorWindow", "Y:"))
        self.ycoord_box.setSuffix(_translate("DetectorWindow", " mm"))
        self.zcoord_label.setText(_translate("DetectorWindow", "Z:"))
        self.zcoord_box.setSuffix(_translate("DetectorWindow", " mm"))
        self.target_label.setText(_translate("DetectorWindow", "Thick Target"))
        self.properties_label.setText(_translate("DetectorWindow", "Width           &         Height"))
        self.width_label.setText(_translate("DetectorWindow", "W:"))
        self.width_box.setSuffix(_translate("DetectorWindow", " mm"))
        self.height_label.setText(_translate("DetectorWindow", "H:"))
        self.height_box.setSuffix(_translate("DetectorWindow", " mm"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    DetectorWindow = QtWidgets.QDialog()
    ui = Ui_DetectorWindow()
    ui.setupUi(DetectorWindow)
    DetectorWindow.show()
    sys.exit(app.exec_())
